### PR TITLE
Add missing sync in vertical volume tendency kernel

### PR DESCRIPTION
**Pull request by mwarusz**
# Description

Fixes a possible race condition in the vertical tendency evaluation.

I have

- [x] Written and run all necessary tests with CLIMA by including `tests/runtests.jl`
- [x] Followed all necessary [style guidelines](https://CliMA.github.io/CLIMA/latest/CodingConventions.html) and run `julia .dev/climaformat.jl .`
- [x] Updated the documentation to reflect changes from this PR.

# For review by CLIMA developers

- [x] There are no open pull requests for this already
- [x] CLIMA developers with relevant expertise have been assigned to review this submission
- [x] The code conforms to the [style guidelines](https://CliMA.github.io/CLIMA/latest/CodingConventions.html) and has consistent naming conventions. `julia .dev/format.jl` has been run in a separate commit.
- [x] This code does what it is technically intended to do (all numerics make sense physically and/or computationally)

### CODE REVIEW

File: src/Numerics/DGmethods/DGmodel_kernels.jl [hunk 1]
```diff
@@ -320,6 +320,9 @@ end
             local_MI[k] = vgeo[ijk, _MI, e]
         end
 
+        # ensure D is loaded
+        @synchronize
```
Comment (jkozdon) [inline src/Numerics/DGmethods/DGmodel_kernels.jl:324]: To match the call on [line 438](https://github.com/CliMA/ClimateMachine.jl/blob/97cbf3b357422a56697bd682755e6676a4a65fef/src/Numerics/DGmethods/DGmodel_kernels.jl#L438) do we want

```suggestion
        @synchronize(dim == 3)
```

Comment (mwarusz) [reply src/Numerics/DGmethods/DGmodel_kernels.jl]: Yes, good catch !

**Pull request merged**